### PR TITLE
Add direction for bindings

### PR DESCRIPTION
**Pull request by yaron2**
Closes https://github.com/dapr/dapr/issues/6449.

### CODE REVIEW

File: tests/config/dapr_kafka_bindings_grpc.yaml [hunk 1]
```diff
@@ -35,3 +35,5 @@ spec:
     value: "false"
   - name: initialOffset
     value: oldest
+  - name: direction
+    value: input, outputs
```
Comment (Taction) [inline tests/config/dapr_kafka_bindings_grpc.yaml:39]: nits: `input, output`

Comment (yaron2) [reply tests/config/dapr_kafka_bindings_grpc.yaml]: Thanks :)

Comment (yaron2) [reply tests/config/dapr_kafka_bindings_grpc.yaml]: Fixed

File: pkg/runtime/runtime.go [hunk 2]
```diff
@@ -1705,7 +1705,30 @@ func (a *DaprRuntime) isAppSubscribedToBinding(binding string) (bool, error) {
 	return false, nil
 }
 
+func isBindingOfDirection(direction string, metadata []componentsV1alpha1.MetadataItem) bool {
+	directionFound := false
+
+	for _, m := range metadata {
+		if strings.EqualFold(m.Name, "direction") {
```
Comment (Taction) [inline pkg/runtime/runtime.go:1712]: Better declare `"direction"` as const.

File: pkg/runtime/runtime.go [hunk 3]
```diff
@@ -1705,7 +1705,30 @@ func (a *DaprRuntime) isAppSubscribedToBinding(binding string) (bool, error) {
...
+				}
+			}
+		}
+	}
+
+	return !directionFound
+}
+
 func (a *DaprRuntime) initInputBinding(c componentsV1alpha1.Component) error {
+	if !isBindingOfDirection("input", c.Spec.Metadata) {
```
Comment (Taction) [inline pkg/runtime/runtime.go:1728]: Same as above.

File: pkg/runtime/runtime.go [hunk 4]
```diff
@@ -1732,6 +1755,10 @@ func (a *DaprRuntime) initInputBinding(c componentsV1alpha1.Component) error {
 }
 
 func (a *DaprRuntime) initOutputBinding(c componentsV1alpha1.Component) error {
+	if !isBindingOfDirection("output", c.Spec.Metadata) {
```
Comment (Taction) [inline pkg/runtime/runtime.go:1758]: Same as above.

**Comment by Taction** [comment]
LGTM, just some nit changes.

**Comment by yaron2** [comment]
> LGTM, just some nit changes.

Thanks, addressed.

**Review by Taction — DISMISSED** [review]
LGTM

**Review by Taction — APPROVED** [review]

**Pull request merged**